### DISCUSSION TITLE
Save params and publish them

### DISCUSSION THREAD

**Discussion by massung** [nextflow-io/nextflow#2892] [Q&A] [answered]
I have a workflow with a considerable number of input parameters.

What I'd like to do is save them to a JSON/YAML file for posterity and have that file published to S3 along with the rest of the workflow's output files so the run can be reproduced in the future if necessary. I currently am using the `exec` directive to create the JSON output, but I cannot publish it since the output is not written to the scratch directory.

Example:

```nextflow
import groovy.json.JsonOutput

params.foo = 'bar'
params.bar = 'baz'

process saveParams {
    output:
        //path 'params.json'

    exec:
        json = JsonOutput.toJson(params)

        paramsFile = new File('params.json')
        paramsFile.write(JsonOutput.prettyPrint(json))
}
```

If I uncomment the output, the pipeline fails saying `params.json` doesn't exist (which it doesn't, it exists next to `main.nf`). Without it the pipeline runs just fine, but I can't publish the output file.

Is there any way for me to do this?

**Comment by jorgeaguileraseqera** [comment, marked as answer]
You can use the `script` section using interpolation expressions. For example:

```
import groovy.json.JsonOutput

params.foo = 'bar'
params.bar = 'baz'

process saveParams {
    output:
        path 'params.json'

    script:
      "echo '${JsonOutput.toJson(params)}' > params.json"
}
```

**Reply by massung** [reply]
Ah, I can't believe I didn't think of that!

Thanks! Hopefully this question will help someone else in the future, too.

**Reply by pmoradi2022** [reply]
It helped me!!!  Thanks guys!
